### ISSUE TITLE
Apparent trouble with EdgeView __contains__

### ISSUE THREAD

**Issue by joelmiller** [networkx/networkx#3819]
See:

https://stackoverflow.com/questions/60184296/how-to-iterate-through-edges-in-a-graph-if-they-are-not-in-a-pre-selected-list

It appears that an EdgeView based on just some of the edges returns `True` for a containment test as long as the edge is in the graph, rather than that it is one of the edges in the EdgeView.

**Comment by dschult** [comment]
Thanks very much for this report. It is indeed a bug in the ```__contains__``` method of the ```EdgeDataView``` family of objects. Let's add the example code from the stackoverflow link as a test. Then fix it, which will involve checking whether either (I think) node is in self._nbunch

```python
graph = nx.Graph()
graph.add_edges_from([(0, 1), (1, 2), (2, 3), (3, 4)])
end_nodes = [n for n in graph.nodes if nx.degree(graph, n) == 1]
end_edges = graph.edges(end_nodes)
print(f"end edges: {end_edges}")
for edge in graph.edges:
    if edge not in end_edges:
        print(f"edge {edge} is not an end edge.")
    else:
        print(f"edge {edge} is an end edge.")
```

**Comment by iboates** [comment]
I am interested in having a try at this. I haven't worked on this library before but I am quite interested in network topology applications. I am also the person who posted the stackoverflow thread which revealed the defect so I feel some attachment and personal interest in resolving it.

**Comment by dschult** [comment]
That sounds great!  Thanks!
The first thing would be to get to know the ```classes/reportviews.py``` objects generally. Probably add your  example that breaks the code as a test in ```classes/tests/test_reportviews.py```.

Then let's nail down the docs on how to handle ```nbunch```.  Right now the EdgeDataView object docstring has very little about what nbunch should do. The last sentence of the docstring is the best we've got. These EdgeDataViews hold a data/default pair of attributes and ```self._nbunch``` which is currently a list of nodes. The object iterates over edges starting from nodes in ```nbunch```. 

We should probably change the ```self._nbunch``` object to be a set of nodes rather than a list because almost everything we do with it is a membership check.

For the ```__contains__``` method, I think nbunch should restrict the view to edges in the graph that start from nodes in nbunch. For ```directed``` graphs you can just check the first node of an edge-tuple. For ```undirected``` graphs, you'd need to check if either node in the edge tuple is in nbunch.   Is this your expected behavior? 

The code to actually enact the fix is in many places: the ```__contains__``` methods of the family of EdgeDataView classes. Start with ```OutEdgeDataView``` and work your way through the classes that build off of that. I hope it just involves adding a check whether ```nbunch is None``` and if not, checking the candidate edge to see if the appropriate node is in nbunch.

Ask questions of course, and feel free to post half-done code as a PR to get feedback... Thanks.

**Comment by rossbar** [comment]
Looks like this was closed by #3845. If not, please reopen.
